app/controllers/OrganizationController.scala

```diff
@@ -0,0 +1,167 @@
+package controllers
+
+import javax.inject.Inject
+import play.api._
+import play.api.mvc._
+import play.api.Play.current
+import play.api.libs.iteratee
+import play.api.libs.json._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import com.mohiva.play.silhouette.api.Authenticator.Implicits._
+import com.mohiva.play.silhouette.api.{Environment, LoginInfo, Silhouette}
+import com.mohiva.play.silhouette.api.exceptions.ProviderException
+import com.mohiva.play.silhouette.impl.authenticators.CookieAuthenticator
+import com.mohiva.play.silhouette.impl.exceptions.IdentityNotFoundException
+import play.api.i18n.{I18nSupport, Messages, MessagesApi}
+import models._
+import models.forms.OrganizationForms
+import models.dispenser._
+import services._
+import java.util.UUID
+
+import play.api.libs.json.{JsPath, Json, OWrites, Reads}
+import utils.authorization.WithRole
+
+class OrganizationController @Inject() (
```

Comment:
Looks like a RESTful controller (`validateJson[...]` in each function), but its authentification is based on our role system implementation using silhouette. Thus, no other micro service will be able to request the data without handling a users session. Alternatively, we have implemented the `ApiAction` class that handles authentification based on creedentials given at any request. 

For now, it is okay to implement it using the role system, especially since there is still an ongoing discussion about potential security issues implemented by a credential based authentification (only secure on an internal network).